spec/ddtrace/contrib/active_storage/tracer_spec.rb

```diff
@@ -0,0 +1,195 @@
+        extend ActiveStorage::Attached::Macros
+      end
+
+      has_one_attached :image
+    end)
+
+    # This is needed to ensure tests write to disk and not make http requests
+    Rails.configuration.active_storage.service_configurations = {
+      local: {
+        service: 'Disk',
+        root: '/tmp/dd-trace-rb/storage'
+      }
+    }
+
+    ActiveRecord::Base.establish_connection(mysql_connection_string)
+
+    begin
+      Article.count
+    rescue ActiveRecord::StatementInvalid
+      ActiveRecord::Schema.define(version: 20180101000000) do
+        create_table 'articles', force: :cascade do |t|
+          t.string   'title'
+          t.datetime 'created_at', null: false
+          t.datetime 'updated_at', null: false
+        end
+
+        create_table :active_storage_blobs do |t|
+          t.string   :key,        null: false
+          t.string   :filename,   null: false
+          t.string   :content_type
+          t.text     :metadata
+          t.bigint   :byte_size,  null: false
+          t.string   :checksum,   null: false
+          t.datetime :created_at, null: false
+
+          t.index [:key], unique: true
+        end
+
+        create_table :active_storage_attachments do |t|
+          t.string     :name,     null: false
+          t.references :record,   null: false, polymorphic: true, index: false
+          t.references :blob,     null: false
+
+          t.datetime :created_at, null: false
+
+          t.index [:record_type, :record_id, :name, :blob_id],
+                  name: 'index_active_storage_attachments_uniqueness',
+                  unique: true
+        end
+      end
+    end
+
+    ApplicationRecord
+  end
+
+  before { app }
+
+  let(:configuration_options) { {} }
+
+  before(:each) do
+    # Prevent extra spans during tests
+    # This is needed to ensure tests write to disk and not make http requests
+    # Rails.configuration.active_storage.service_configurations = {
+    #   local: {
+    #     service: 'Disk',
+    #     root: '/tmp/dd-trace-rb/storage'
+    #   }
+    # }
+
+    Datadog.configure do |c|
+      c.use :active_storage, configuration_options
+    end
+  end
+
+  around do |example|
+    # Reset before and after each example; don't allow global state to linger.
+    ClimateControl.modify('USE_ACTIVE_STORAGE' => 'true') do
+      Datadog.registry[:active_storage].reset_configuration!
+      example.run
+      Datadog.registry[:active_storage].reset_configuration!
+    end
+  end
+
+  # From:
+  # https://github.com/rails/rails/blob/2a3f8a29e5d7d17a0448fa47fd756c8d11b175b1/activestorage/app/models/active_storage/blob.rb#L64
+  # https://github.com/rails/rails/blob/06c94818d640d13efecc722c4ec8c9c5ea2a99b0/activestorage/app/models/active_storage/blob.rb#L104
+  def create_blob(data: 'Hello world!', filename: 'hello.txt', content_type: 'text/plain', identify: true, record: nil)
+    if Rails.version >= '6.0'
```

Comment:
Similar question/suggestion about possibly moving this to https://github.com/DataDog/dd-trace-rb/tree/master/test/contrib/rails/apps.

Reply:
yea i think that's reasonable